core/perf_test/CMakeLists.txt

```diff
@@ -114,3 +114,91 @@ IF(NOT Kokkos_ENABLE_OPENMPTARGET)
     CATEGORIES PERFORMANCE
   )
 ENDIF()
+
+
+IF(NOT Kokkos_ENABLE_BENCHMARKS OR KOKKOS_HAS_TRILINOS)
+  RETURN()
+ENDIF()
+
+find_package(benchmark QUIET)
+IF(NOT benchmark_FOUND)
+  message(STATUS "No installed benchmark found, fetching from GitHub")
+  include(FetchContent)
+  SET(BENCHMARK_ENABLE_TESTING OFF)
+
+  FetchContent_Declare(
+        googlebenchmark
+        GIT_REPOSITORY https://github.com/google/benchmark.git
+        GIT_TAG v1.6.2)
+
+  FetchContent_MakeAvailable(
+        googlebenchmark)
+
+  include_directories(${benchmark_SOURCE_DIR}/include)
+
+  # Suppress clang-tidy diagnostics on code that we do not have control over
+  IF(CMAKE_CXX_CLANG_TIDY)
+    SET_TARGET_PROPERTIES(benchmark PROPERTIES CXX_CLANG_TIDY "")
+  ENDIF()
+
+  target_compile_options(benchmark PRIVATE -w)
+  target_compile_options(benchmark_main PRIVATE -w)
+ENDIF()
+
+
+FUNCTION(KOKKOS_ADD_BENCHMARK NAME)
+  CMAKE_PARSE_ARGUMENTS(
+    BENCHMARK
+    ""
+    ""
+    "SOURCES"
+    ${ARGN}
+  )
+  IF(DEFINED BENCHMARK_UNPARSED_ARGUMENTS)
+    MESSAGE(
+      WARNING
+      "Unexpected arguments when adding a benchmark: "
+      ${BENCHMARK_UNPARSED_ARGUMENTS}
+    )
+  ENDIF()
+
+  SET(BENCHMARK_NAME ${PACKAGE_NAME}_${NAME})
+
+  ADD_EXECUTABLE(
+    ${BENCHMARK_NAME}
+    ${BENCHMARK_SOURCES}
+  )
+  TARGET_LINK_LIBRARIES(
+    ${BENCHMARK_NAME}
+    PRIVATE benchmark::benchmark Kokkos::kokkos
+  )
+  FOREACH(SOURCE_FILE ${BENCHMARK_SOURCES})
+    SET_SOURCE_FILES_PROPERTIES(
+      ${SOURCE_FILE}
+      PROPERTIES LANGUAGE ${KOKKOS_COMPILE_LANGUAGE}
+    )
+  ENDFOREACH()
+
+  STRING(TIMESTAMP BENCHMARK_TIME "%Y-%m-%d_T%H-%M-%S" UTC)
+  SET(
+    BENCHMARK_ARGS
+    --benchmark_counters_tabular=true
+    --benchmark_out=${BENCHMARK_NAME}_${BENCHMARK_TIME}.json
+  )
```

Comment:
@dalg24 Turns out ~40 lines of pure CMake is enough to implement that.

Reply:
As I see this ... could we also use the machine name we do the benchmark on in the filename? might be helpful when comparing stuff

Reply:
`benchmark` lib already puts it in the output, see JSON for example:
```json
{
  "context": {
    "date": "2022-08-17T18:30:06+02:00",
    "host_name": "rysy",
    "executable": "./build/core/perf_test/KokkosCore_PerformanceTest_Benchmark",
    "num_cpus": 8,
    "mhz_per_cpu": 4700,
(...)
}

Reply:
perfect